Source/WebCore/rendering/svg/SVGInlineTextBox.cpp

```diff
@@ -669,6 +669,17 @@ void SVGInlineTextBox::paintTextWithShadows(GraphicsContext& context, const Rend
                     usedContext->setCompositeOperation(CompositeOperator::DestinationIn);
                     usedContext->beginTransparencyLayer(1);
                 }
+
+                if (paintingResourceMode().contains(RenderSVGResourceMode::ApplyToFill) && usedContext->fillPattern()) {
+                    usedContext->fillRect(FloatRect::infiniteRect());
+                    usedContext->setCompositeOperation(CompositeOperator::DestinationIn);
+                    usedContext->beginTransparencyLayer(1);
+                } else if (usedContext->strokePattern()) {
+                    usedContext->setFillPattern(*usedContext->strokePattern());
```

Comment:
I think you also want to extend GraphicsContext :-) a `fillRect(const FloatRect&, Pattern&)` variant, would allow you to avoid the state mutation here. (Note: you also need to assure it's reset correctly, the current restore() time is too lat probably)